### PR TITLE
[Fix-12339][Api] Fix optional query parameter miss default value issue #12339

### PR DESCRIPTION
**Pull request by zuzuviewer**
Fix optional query parameter miss default value issue

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/SchedulerController.java [hunk 1]
```diff
@@ -158,7 +158,7 @@ public Result updateSchedule(@ApiIgnore @RequestAttribute(value = SESSION_USER)
                                  @PathVariable(value = "id") Integer id,
                                  @RequestParam(value = "schedule") String schedule,
                                  @RequestParam(value = "warningType", required = false, defaultValue = DEFAULT_WARNING_TYPE) WarningType warningType,
-                                 @RequestParam(value = "warningGroupId", required = false) int warningGroupId,
+                                 @RequestParam(value = "warningGroupId", required = false, defaultValue = DEFAULT_NOTIFY_GROUP_ID) int warningGroupId,
```
Comment (fuchanghai) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/SchedulerController.java:161]: if value of ```warningType``` is ```WarningType .NONE``` , value of ```warningGroupId``` should be null. so i don't think ```warningGroupId``` be point default value. maybe we should change the type of ```warningGroupId``` to Integer .WDYT? cc @SbloodyS @EricGao888 @caishunfeng

**Review by fuchanghai — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/SchedulerController.java (hunk 1 above)
Comment (fuchanghai) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/SchedulerController.java:161]: > if value of `warningType` is `WarningType .NONE` , value of `warningGroupId` should be null. so i don't think `warningGroupId` be point default value. maybe we should change the type of `warningGroupId` to Integer .WDYT? cc @SbloodyS @EricGao888 @caishunfeng

this seems like had been fixed #11774

Comment (zuzuviewer) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/SchedulerController.java:161]: Please note different file name, this is 'SchedulerController.java", that is 'ExecutorController.java'

Comment (fuchanghai) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/SchedulerController.java:161]: > Please note different file name, this is 'SchedulerController.java", that is 'ExecutorController.java'

sorry ,but  i don't think ```warningGroupId``` should  be pointed default value

Comment (fuchanghai) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/SchedulerController.java:161]: > if value of `warningType` is `WarningType .NONE` , value of `warningGroupId` should be null. so i don't think `warningGroupId` be point default value. maybe we should change the type of `warningGroupId` to Integer .WDYT? cc @SbloodyS @EricGao888 @caishunfeng

@zuzuviewer  i think  change the type of ```warningGroupId``` to Integer is a good way , WDYT? cc @SbloodyS @EricGao888

Comment (zhongjiajie) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/SchedulerController.java:161]: > if value of `warningType` is `WarningType .NONE` , value of `warningGroupId` should be null. so i don't think `warningGroupId` be point default value. maybe we should change the type of `warningGroupId` to Integer .WDYT? cc @SbloodyS @EricGao888 @caishunfeng

I agree with that, we should still make `warningGroupId` optional

**Pull request closed without merging**